frappe/public/js/frappe/request.js

```diff
@@ -187,12 +178,23 @@ frappe.request.call = function(opts) {
 				frappe.request.report_error(xhr, opts);
 			}
 		},
+		501: function(data, xhr) {
+			if(typeof data === "string") data = JSON.parse(data);
+			opts.error_callback && opts.error_callback(data, xhr.responseText);
+		},
+		502: function (xhr) {
+			frappe.msgprint(__("Internal Server Error"));
+		},
 		504: function(xhr) {
 			frappe.msgprint(__("Request Timed Out"))
 			opts.error_callback && opts.error_callback();
 		},
-		502: function(xhr) {
-			frappe.msgprint(__("Internal Server Error"));
+		508: function (xhr) {
+			frappe.utils.play_sound("error");
+			frappe.msgprint({
+				title: __('Please try again'), indicator: 'red',
+				message: __("Another transaction is blocking this one. Please try again in a few seconds.")
```

Comment:
Where did this come from?

Reply:
It's just restructuring and in ascending order. The 508 was on the top of the changes here.